File: backend/api/src/main/java/com/github/ai/leetcodequiz/api/QuestionnaireItemDto.java
package com.github.ai.leetcodequiz.api;

import org.jetbrains.annotations.NotNull;

import java.util.List;

public record QuestionnaireItemDto(
        @NotNull String id,
        boolean isComplete,
        @NotNull List<QuestionItemDto> nextQuestions
) {
}

File: backend/api/src/main/scala/com/github/ai/leetcodequiz/api/QuestionnaireItemDto.scala
package com.github.ai.leetcodequiz.api

import zio.json.{JsonDecoder, JsonEncoder}

case class QuestionnaireItemDto(
  id: String,
  isComplete: Boolean,
  questions: List[QuestionItemDto],
  answers: List[QuestionAnswerDto],
  stats: QuestionnaireStatsDto
) derives JsonEncoder,
      JsonDecoder
